returnn/tf/layers/rec.py

```diff
@@ -1082,6 +1085,8 @@ def collect_exception(cls, layer_name):
           else:
             out = StringIO()
             better_exchook.better_exchook(exc_type, value, tb, file=out)
+            if isinstance(value, fail_directly_exception_types):
+              raise
```

Comment:
You should move this two lines above. The `out` and `better_exchook` is not needed.